lib/proxy.ts

```diff
@@ -1,10 +1,24 @@
 import { ethers } from 'ethers';
+import proxyABI from './proxy-abi.json';
 
 
-export function getProxy(signer: ethers.JsonRpcSigner) {
-  return new ethers.Contract(
-    process.env.NEXT_PUBLIC_PROXY_CONTRACT_ADDRESS!,
-    ["function attestByDelegation(string id, (bytes32 schema, (address recipient, uint64 expirationTime, bool revocable, bytes32 refUID, bytes data, uint256 value) data, (uint8 v, bytes32 r, bytes32 s) signature, address attester, uint64 deadline) delegatedRequest)"],
-    signer
-  )
+export function getProxy(signer?: ethers.JsonRpcSigner, provider?: ethers.JsonRpcProvider) {
```

Comment:
It seems you are passing the signer or provider optional argument to the `ethers.Contract` constructor.

A simpler way to do the same (and doesn't require any type checking) is:

```ts
export function getProxy(signerOrProvider?: ethers.JsonRpcSigner | ethers.JsonRpcProvider) {
  return new ethers.Contract(
    process.env.NEXT_PUBLIC_PROXY_CONTRACT_ADDRESS!,
    proxyABI,
    signerOrProvider
}
```

Reply:
Thanks, nice tip.